src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacPropertyAttributes/DacPropertyAttributesAnalyzer.cs

```diff
@@ -44,66 +48,173 @@ private static Task AnalyzePropertyAsync(SymbolAnalysisContext symbolContext, PX
+			{
+				return true;
+			}
+
+			if (specialAttributesDeclaredOnDacProperty.Count > 1)
+			{
+				await RegisterDiagnosticForAttributesAsync(symbolContext, specialAttributesDeclaredOnDacProperty,
+														   Descriptors.PX1023_MultipleSpecialTypeAttributesOnProperty)
+													  .ConfigureAwait(false);
+			}
+
+			if (specialAttributesWithConflictingAggregatorDeclarations.Count > 0)
+			{
+				await RegisterDiagnosticForAttributesAsync(symbolContext, specialAttributesDeclaredOnDacProperty,
+														   Descriptors.PX1023_MultipleSpecialTypeAttributesOnAggregators)
+													  .ConfigureAwait(false);
+			}
+
+			return false;
+
+			//-----------------------------------------------Local Functions---------------------------------------
+			(List<AttributeData>, List<AttributeData>) FilterSpecialAttributes()
+			{
+				List<AttributeData> specialAttributesOnDacProperty = new List<AttributeData>(2);
+				List<AttributeData> specialAttributesInvalidAggregatorDeclarations = new List<AttributeData>(2);
+
+				foreach (var attrWithInfos in attributesWithInfos)
+				{
+					int countOfSpecialAttributeInfos = attrWithInfos.Infos.Count(atrInfo => atrInfo.IsSpecial);
+
+					if (countOfSpecialAttributeInfos > 0)
+					{
+						specialAttributesOnDacProperty.Add(attrWithInfos.Attribute);
+					}
+
+					if (countOfSpecialAttributeInfos > 1)
+					{
+						specialAttributesInvalidAggregatorDeclarations.Add(attrWithInfos.Attribute);
+					}
+				}
+
+				return (specialAttributesOnDacProperty, specialAttributesInvalidAggregatorDeclarations);
+			}
+		}
+
+		private static async Task CheckForFieldTypeAttributesAsync(IPropertySymbol property, SymbolAnalysisContext symbolContext, PXContext pxContext,
+														List<(AttributeData Attribute, List<FieldTypeAttributeInfo> Infos)> attributesWithInfos)
+		{
+			var (typeAttributesDeclaredOnDacProperty, typeAttributesWithConflictingAggregatorDeclarations) = FilterTypeAttributes();
+
+			if (typeAttributesDeclaredOnDacProperty.Count == 0)
+				return;
+
+			if (typeAttributesWithConflictingAggregatorDeclarations.Count > 0)
+			{
+				await RegisterDiagnosticForAttributesAsync(symbolContext, typeAttributesWithConflictingAggregatorDeclarations,
+														   Descriptors.PX1023_MultipleTypeAttributesOnAggregators)
+													  .ConfigureAwait(false);
+			}
+
+			if (typeAttributesDeclaredOnDacProperty.Count > 1)					
+			{
+				await RegisterDiagnosticForAttributesAsync(symbolContext, typeAttributesDeclaredOnDacProperty,
+														   Descriptors.PX1023_MultipleTypeAttributesOnProperty)
+													  .ConfigureAwait(false);
+			} 
+			else if (typeAttributesWithConflictingAggregatorDeclarations.Count == 0)
+			{
+				var typeAttribute = typeAttributesDeclaredOnDacProperty[0];
+				var fieldType = attributesWithInfos.First(attrWithInfo => attrWithInfo.Attribute.Equals(typeAttribute))
```

Comment:
Are you sure that `attributesWithInfos` is not empty at all times?

Reply:
Yes, 100% sure because `typeAttributesDeclaredOnDacProperty` collection is a filtered subset of the `attributesWithInfos` collection.  In this case it contains only one element and it should be present in the `attributesWithInfos`

Reply:
@SENya1990 ok